lib/ddtrace/tracer.rb

```diff
@@ -288,7 +288,15 @@ def trace(name, options = {})
         # It's not a problem since we re-raise it afterwards so for example a
         # SignalException::Interrupt would still bubble up.
         rescue Exception => e
-          (options[:on_error] || DEFAULT_ON_ERROR).call(span, e)
+          valid_error = options[:on_error] || DEFAULT_ON_ERROR ## (options[:on_error] || DEFAULT_ON_ERROR).call(span, e)
+
+          if valid_error.respond_to?(:call) # && valid_error.method(:call).arity == 2 // we can use this to check that 2 and only two arguments are being passed
```

Comment:
I'm ok with skipping the arity check.
As long as we are provided a callable we should go ahead and try to call with with the two arguments.